lib/matplotlib/pyplot.py

```diff
@@ -426,43 +426,43 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
         `num`.
 
     figsize : tuple of integers, optional, default: None
-        width, height in inches. If not provided, defaults to rc
-        figure.figsize.
+        width, height in inches. If not provided, defaults to 
+        :rc:`figure.figsize`.
 
     dpi : integer, optional, default: None
-        resolution of the figure. If not provided, defaults to rc figure.dpi.
+        resolution of the figure. If not provided, defaults to :rc:`figure.dpi`.
 
     facecolor :
-        the background color. If not provided, defaults to rc figure.facecolor.
+        the background color. If not provided, defaults to :rc:`figure.facecolor`.
 
     edgecolor :
-        the border color. If not provided, defaults to rc figure.edgecolor.
+        the border color. If not provided, defaults to :rc:`figure.edgecolor`.
 
     frameon : bool, optional, default: True
         If False, suppress drawing the figure frame.
 
-    FigureClass : class derived from matplotlib.figure.Figure
+    FigureClass : class derived from the :class:`~matplotlib.figure.Figure` class.
         Optionally use a custom Figure instance.
 
     clear : bool, optional, default: False
         If True and the figure already exists, then it is cleared.
 
     Returns
     -------
-    figure : Figure
+    figure : :class:`~matplotlib.figure.Figure` object
         The Figure instance returned will also be passed to new_figure_manager
         in the backends, which allows to hook custom Figure classes into the
-        pylab interface. Additional kwargs will be passed to the figure init
-        function.
+        pylab interface. Additional kwargs will be passed to the 
+        :class:`~matplotlib.figure.Figure` init function.
```

Comment:
We use the sphinx default role 'obj' so this can be simplified. No need for the Prefix. Within text blocks, we use the shortest form possible.

\`~matplotlib.figure.Figure\` or maybe even \`.Figure\` is enough.

Reply:
I have mostly looked at the other docstrings in regard to prefixes. Are there any directives for how  to use them (couldn't find anything in the documentation)? :func: and :meth: adds parenthesis but are there any other reasons to use them?

Reply:
Unfortunately, the docs on documentation is not up-to-date in that respect. See *Citing other types* in https://github.com/matplotlib/matplotlib/issues/10225.

That using the default role does not add parentheses is a bug in Sphinx: https://github.com/sphinx-doc/sphinx/issues/4409 I wouldn't make that a criterion how to write our docstrings.

Reply:
I wanted to know if we should use for example 
:class: `.Figure` or just `.Figure`

Should :class:, :meth:, :func:, :mod: etc be used or not? I haven't found anything about this anywhere but some docstrings include those but I believe that many don't.

Reply:
*Citing other types* in https://github.com/matplotlib/matplotlib/issues/10225 is the current proposal.

Don't rely too much on what's currently present in the docs. We have a lot of legacy code that has not yet been updated to and still uses ``:class:`.Figure` `` everywhere.

Reply:
Ok, so :class: etc should not be used? I found that section confusing  when it seems to only talk about when to use  ~matplotlib.markers.MarkerStyle or .MarkerStyle and not anything about :class: etc.

Reply:
Oh sorry, I didn't read carefully.

:class: is basically almost never needed (except for rare cases where name may not be unique otherwise, e.g. `cycler`).

Reply:
I updated the code.

I still don't understand what is the best practice, convention and what works in regard to :func:, :class: etc. 

I need to use :func: and :meth: in places where I don't understand why. It seems to not look outside of the actual function or method without them when searching for the function or method even though I even used the full link.